lib/spack/spack/spec.py

```diff
@@ -938,6 +938,7 @@ def __init__(self, spec_like=None,
         self.namespace = None
 
         self._hash = None
+        self._build_hash = None
```

Comment:
There's already a field for `_full_hash` below (it should likely be moved up here to go with the other hashes).  `_build_hash` is redundant.